tests/Feature/Repositories/ShoppingCartTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace IndieHD\Velkart\Tests\Feature\Repositories;
+
+use IndieHD\Velkart\Contracts\OrderRepositoryContract;
+use IndieHD\Velkart\Contracts\ShoppingCartRepositoryContract;
+use Ramsey\Uuid\Uuid;
```

Comment:
Is the dependency inject-able using a contract by chance ? This is definitely an implementation that has better odds of being swapped out later so it may make since to use it in such a way.

Reply:
Good question. I had the same thought initially and I did look into it but it wasn't straightforward because there seems to be a constructor that requires arguments:

```
Illuminate\Contracts\Container\BindingResolutionException : Unresolvable dependency resolving [Parameter #0 [ <required> array $fields ]] in class Ramsey\Uuid\Uuid
```

I'll dig a little deeper.

Reply:
I just pushed a correction for this. Thanks again.

Reply:
FYI it is available through `Illuminate\Support\Str` — e.g. `Str::uuid()` but I understand if you'd want it injected.

Reply:
@mblarsen

I think he's using the 3rd party UUID package for other reasons then just generating unique id's ? Forgive me if I sound ignorant. Its been a minute ;)